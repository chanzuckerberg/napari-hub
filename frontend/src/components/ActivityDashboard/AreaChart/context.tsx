import { createContext, ReactNode, useContext, useMemo } from 'react';

interface AreaChartStateValue {
  publicReleaseX: number;
  isScreen600: boolean;
  visibleMonths: number[];
}

const AreaChartStateContext = createContext<AreaChartStateValue>({
  publicReleaseX: 0,
  isScreen600: false,
  visibleMonths: [],
});

interface Props extends AreaChartStateValue {
  children: ReactNode;
}

/**
 * Provider for sharing state used in the area chart component.
 */
export function AreaChartStateProvider({
  children,
  isScreen600,
  publicReleaseX,
  visibleMonths,
}: Props) {
  const value = useMemo<AreaChartStateValue>(
    () => ({
      publicReleaseX,
      isScreen600,
      visibleMonths,
    }),
    [isScreen600, publicReleaseX, visibleMonths],
  );

  return (
    <AreaChartStateContext.Provider value={value}>
      {children}
    </AreaChartStateContext.Provider>
  );
}

export function useAreaChartState(): AreaChartStateValue {
  return useContext(AreaChartStateContext);
}
