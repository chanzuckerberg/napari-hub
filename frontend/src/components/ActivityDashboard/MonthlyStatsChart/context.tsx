import dayjs from 'dayjs';
import { createContext, ReactNode, useContext, useMemo } from 'react';

import { I18nKeys } from '@/types/i18n';

interface MonthlyStatsValue {
  dateLineI18nKey: I18nKeys<'activity'>;
  startDate: dayjs.ConfigType;
}

const MonthlyStatsContext = createContext<MonthlyStatsValue | null>(null);

interface Props
  extends Pick<MonthlyStatsValue, 'startDate' | 'dateLineI18nKey'> {
  children: ReactNode;
}

export function MonthlyStatsProvider({
  children,
  dateLineI18nKey,
  startDate,
}: Props) {
  const value = useMemo(
    () => ({
      dateLineI18nKey,
      startDate,
    }),
    [dateLineI18nKey, startDate],
  );

  return (
    <MonthlyStatsContext.Provider value={value}>
      {children}
    </MonthlyStatsContext.Provider>
  );
}

export function useMonthlyStats(): MonthlyStatsValue {
  const value = useContext(MonthlyStatsContext);

  if (!value) {
    throw new Error('useMonthlyStats() must be used in a MonthlyStatsProvider');
  }

  return value;
}
