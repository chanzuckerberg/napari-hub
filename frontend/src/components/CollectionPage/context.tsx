import { createContext, ReactNode, useContext } from 'react';

import { CollectionData } from '@/types/collections';

const CollectionContext = createContext<CollectionData | null>(null);

interface Props {
  collection: CollectionData;
  children: ReactNode;
}

/**
 * Provider for access to collection state from anywhere in collection page.
 */
export function CollectionContextProvider({ collection, children }: Props) {
  return (
    <CollectionContext.Provider value={collection}>
      {children}
    </CollectionContext.Provider>
  );
}

/**
 * Hook for accessing collection data from hub API.
 */
export function useCollection(): CollectionData {
  const collection = useContext(CollectionContext);

  if (!collection) {
    throw new Error(
      'useCollections() may not be used outside of CollectionContextProvider',
    );
  }

  return collection;
}
