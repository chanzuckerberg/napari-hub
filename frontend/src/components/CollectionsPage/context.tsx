import { createContext, ReactNode, useContext } from 'react';

import { CollectionIndexData } from '@/types/collections';

const CollectionsContext = createContext<CollectionIndexData[]>([]);

interface Props {
  collections: CollectionIndexData[];
  children: ReactNode;
}

/**
 * Provider for access to collections state from anywhere in collections home
 * page UI.
 */
export function CollectionsContextProvider({ collections, children }: Props) {
  return (
    <CollectionsContext.Provider value={collections}>
      {children}
    </CollectionsContext.Provider>
  );
}

/**
 * Hook for accessing list of collections from hub API.
 */
export function useCollections(): CollectionIndexData[] {
  return useContext(CollectionsContext);
}
