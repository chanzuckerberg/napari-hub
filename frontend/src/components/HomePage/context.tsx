import { createContext, ReactNode, useContext, useMemo } from 'react';

import { PluginSectionsResponse } from '@/types';

interface HomePageContextValue {
  pluginSections: PluginSectionsResponse;
}

const HomePageContext = createContext<HomePageContextValue | null>(null);

interface Props extends Pick<HomePageContextValue, 'pluginSections'> {
  children: ReactNode;
}

export function HomePageProvider({ children, pluginSections }: Props) {
  const value = useMemo(
    () => ({
      pluginSections,
    }),
    [pluginSections],
  );

  return (
    <HomePageContext.Provider value={value}>
      {children}
    </HomePageContext.Provider>
  );
}

export function useHomePage(): HomePageContextValue {
  const value = useContext(HomePageContext);

  if (!value) {
    throw new Error('useHomePage must be used in a HomePageProvider');
  }

  return value;
}
