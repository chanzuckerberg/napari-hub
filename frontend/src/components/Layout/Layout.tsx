import { ReactNode } from 'react';

import { AppBar } from '@/components/AppBar';
import { Footer } from '@/components/Footer';
import { SignupForm } from '@/components/SignupForm';

interface Props {
  children: ReactNode;
}

export function Layout({ children }: Props) {
  return (
    <div className="flex flex-col min-h-screen">
      {!process.env.PREVIEW && <AppBar />}

      <main className="flex flex-col flex-grow">{children}</main>
      <SignupForm />
      <Footer />
    </div>
  );
}
