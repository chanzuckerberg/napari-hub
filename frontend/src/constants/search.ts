/**
 * ID used for the plugin search bar on the home page.
 */
export const SEARCH_BAR_ID = 'search-bar';
