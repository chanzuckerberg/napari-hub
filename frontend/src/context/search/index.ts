export * from './constants';
export * from './search.context';
export type { SearchResult, SearchResultMatch } from './search.types';
