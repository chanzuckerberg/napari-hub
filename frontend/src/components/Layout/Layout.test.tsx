import { render, RenderResult, screen } from '@testing-library/react';
import { ReactNode } from 'react';

import { SearchStoreProvider } from '@/store/search/context';

import { Layout } from './Layout';

describe('<Layout />', () => {
  const text = 'Hello, World!';
  let component: RenderResult;
  let child: ReactNode;

  beforeEach(() => {
    child = <h1>{text}</h1>;
    component = render(
      <SearchStoreProvider>
        <Layout>{child}</Layout>
      </SearchStoreProvider>,
    );
  });

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render child', () => {
    const renderedChild = screen.getByText(text);
    expect(renderedChild).toBeTruthy();
  });
});
