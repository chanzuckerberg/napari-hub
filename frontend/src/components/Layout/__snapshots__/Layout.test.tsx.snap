// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col min-h-screen"
  >
    <div
      class="fresnel-container fresnel-lessThan-screen-600 "
    />
    <header
      class="bg-napari-primary h-napari-app-bar justify-center items-center px-6 screen-495:px-12 screen-1150:p-0 zero:grid-cols-[min-content,1fr] grid justify-center gap-6 md:gap-12 grid-cols-2 screen-875:grid-cols-napari-3 screen-1150:grid-cols-napari-4 screen-1425:grid-cols-napari-5"
    >
      <div
        class="fresnel-container fresnel-greaterThanOrEqual-screen-600 "
      >
        <nav
          class="flex"
        >
          <a
            class="whitespace-nowrap"
            data-testid="appBarHome"
            href="/"
          >
            napari &lt;strong&gt;hub&lt;/strong&gt;
          </a>
          <div
            class="flex"
          >
            <a
              class="ml-6"
              href="/about"
            >
              About
            </a>
            <a
              class="ml-6"
              href="/faq"
            >
              FAQ
            </a>
          </div>
        </nav>
      </div>
      <div
        class="fresnel-container fresnel-lessThan-screen-600 "
      >
        <nav
          class="flex"
        >
          <a
            class="whitespace-nowrap"
            data-testid="appBarHome"
            href="/"
          >
            napari &lt;strong&gt;hub&lt;/strong&gt;
          </a>
        </nav>
      </div>
      <div
        class="flex items-center w-full justify-self-end screen-875:col-span-2 screen-1150:col-span-3"
      >
        <form
          class="flex flex-auto items-center border-b-2 border-black"
          data-testid="searchBarForm"
        >
          <input
            aria-label="Search bar for searching for napari plugins."
            class="flex flex-auto border-none outline-none bg-transparent w-0"
            data-testid="searchBarInput"
            value=""
          />
          <button
            aria-label="Submit search query"
            class="MuiButtonBase-root MuiIconButton-root"
            data-testid="submitQueryButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                class="h-5 w-5"
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8.32004 7.68001L0.746704 15.2533M10.6667 9.60001C13.0134 9.60001 14.9334 7.68001 14.9334 5.33335C14.9334 2.98668 13.0134 1.06668 10.6667 1.06668C8.32004 1.06668 6.40004 2.98668 6.40004 5.33335C6.40004 7.68001 8.32004 9.60001 10.6667 9.60001Z"
                  stroke="black"
                  stroke-width="2"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
        <div
          class="fresnel-container fresnel-lessThan-screen-600 ml-6 flex"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Icon for opening side menu.
                </title>
                <path
                  d="M16 1H0"
                  stroke="black"
                  stroke-width="1.98857"
                />
                <path
                  d="M16 7.99924H0"
                  stroke="black"
                  stroke-width="1.98857"
                />
                <path
                  d="M16 15H0"
                  stroke="black"
                  stroke-width="1.98857"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
    <main
      class="flex flex-col flex-grow"
    >
      <h1>
        Hello, World!
      </h1>
    </main>
    <div
      class="bg-napari-light p-6 screen-495:p-12 grid justify-center gap-6 md:gap-12 grid-cols-2 screen-875:grid-cols-napari-3 screen-1150:grid-cols-napari-4 screen-1425:grid-cols-napari-5"
    >
      <div
        class="col-span-2 screen-495:col-span-3 screen-1425:col-start-2"
      >
        <h3
          class="text-lg font-semibold mb-1 screen-495:mb-4"
        >
          Sign up to receive updates
        </h3>
        <form
          action="https://dev.us5.list-manage.com/subscribe/post?u=79f50c0d0474ff43dffedabda&id=63bf16756e"
          class="grid grid-cols-2 screen-875:grid-cols-napari-3 screen-495:gap-x-12 gap-y-3.5"
          method="post"
          novalidate=""
          target="_blank"
        >
          <div
            aria-label="Email address"
            class="MuiFormControl-root MuiTextField-root text-sm pt-0.5 h-8 screen-495:h-[35px] col-span-2 screen-495:col-span-1 screen-875:col-span-2"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                data-testid="emailField"
                name="EMAIL"
                placeholder="email"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained text-sm font-semibold h-[35px] col-span-2 screen-495:col-span-1 MuiButton-containedPrimary MuiButton-disableElevation"
            data-testid="submitButton"
            name="subscribe"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Subscribe
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
    <div
      class="flex flex-col screen-655:flex-row justify-items-center bg-napari-primary p-6 screen-495:px-12"
    >
      <div
        class="w-full flex-1 flex flex-row items-center justify-start mb-6 screen-655:mb-0 screen-655:mr-12"
      >
        <a
          class="whitespace-nowrap font-semibold text-sm mr-6 last:mr-0"
          href="/about"
        >
          About
        </a>
        <a
          class="whitespace-nowrap font-semibold text-sm mr-6 last:mr-0"
          href="/faq"
        >
          FAQ
        </a>
        <a
          class="whitespace-nowrap font-semibold text-sm mr-6 last:mr-0"
          href="/privacy"
        >
          Privacy
        </a>
        <a
          class="whitespace-nowrap font-semibold text-sm mr-6 last:mr-0"
          href="/contact"
        >
          Contact
        </a>
        <a
          class="whitespace-nowrap font-semibold text-sm flex flex-row items-center mr-6 last:mr-0"
          href="https://github.com/chanzuckerberg/napari-hub"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="inline-block mr-0.5 h-3 w-3"
            fill="none"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 0.173252C3.1325 0.173252 0 3.3075 0 7.17325C0 10.2667 2.0055 12.8899 4.78625 13.8145C5.13625 13.8804 5.26458 13.664 5.26458 13.4779C5.26458 13.3117 5.25875 12.8713 5.25583 12.2879C3.30867 12.7103 2.898 11.3488 2.898 11.3488C2.5795 10.5408 2.11925 10.325 2.11925 10.325C1.48517 9.891 2.16825 9.89975 2.16825 9.89975C2.87117 9.94875 3.24042 10.6208 3.24042 10.6208C3.86458 11.6912 4.879 11.382 5.27917 11.2029C5.34217 10.7503 5.52242 10.4417 5.7225 10.2667C4.16792 10.0917 2.534 9.48967 2.534 6.8075C2.534 6.04334 2.80525 5.41917 3.25442 4.92917C3.17567 4.75242 2.93942 4.04075 3.31567 3.0765C3.31567 3.0765 3.90192 2.88867 5.24067 3.794C5.80067 3.63825 6.39567 3.56125 6.99067 3.55775C7.58567 3.56125 8.18067 3.63825 8.74067 3.794C10.0707 2.88867 10.6569 3.0765 10.6569 3.0765C11.0332 4.04075 10.7969 4.75242 10.7269 4.92917C11.1732 5.41917 11.4444 6.04334 11.4444 6.8075C11.4444 9.49667 9.80817 10.0888 8.25067 10.2608C8.49567 10.4708 8.72317 10.9002 8.72317 11.5558C8.72317 12.4927 8.71442 13.2452 8.71442 13.4727C8.71442 13.6564 8.83692 13.8752 9.19567 13.8052C11.9962 12.887 14 10.262 14 7.17325C14 3.3075 10.8657 0.173252 7 0.173252Z"
              fill="black"
            />
          </svg>
          GitHub repo
        </a>
      </div>
      <div
        class="flex flex-row flex-grow w-min justify-end self-end"
      >
        <a
          class="mr-2"
          href="https://napari.org"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="w-8 h-8"
            fill="none"
            height="30"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Go to napari main website.
            </title>
            <path
              d="M27.8387 0H2.16127C0.967636 0 0 0.967636 0 2.16127V27.8387C0 29.0324 0.967636 30 2.16127 30H27.8387C29.0324 30 30 29.0324 30 27.8387V2.16127C30 0.967636 29.0324 0 27.8387 0Z"
              fill="#26283D"
            />
            <path
              d="M27.7779 1.03139H2.22356C1.56514 1.03139 1.03139 1.56514 1.03139 2.22357V27.7779C1.03139 28.4363 1.56514 28.9701 2.22356 28.9701H27.7779C28.4363 28.9701 28.9701 28.4363 28.9701 27.7779V2.22357C28.9701 1.56514 28.4363 1.03139 27.7779 1.03139Z"
              fill="url(#paint0_radial)"
            />
            <path
              d="M17.4655 27.3361C17.099 27.3361 16.7542 27.3201 16.4124 27.2897C16.0096 27.2645 15.61 27.202 15.2187 27.1028C14.4296 26.8759 13.6918 26.4986 13.0459 25.9918C12.1394 25.2971 11.3826 24.4265 10.8209 23.4322C10.5949 23.0338 10.3863 22.4775 9.83293 20.9899C9.71125 20.661 9.61999 20.4104 9.54321 20.2091C9.36359 19.7209 9.3042 19.5558 9.15789 19.2385C8.96705 18.7853 8.73274 18.3516 8.45823 17.9435C8.12277 17.4768 7.73384 17.0509 7.29937 16.6745C7.13279 16.5297 7.05601 16.4616 6.91405 16.3515C6.73732 16.2067 6.49396 16.0183 6.02752 15.6098C5.2395 14.9174 4.94254 14.5958 4.66876 14.2453C4.25597 13.7417 3.92726 13.1748 3.69532 12.5664C3.49242 12.0272 3.37416 11.4599 3.34476 10.8846C3.32978 10.3829 3.38289 9.88145 3.50265 9.39401C3.69467 8.45513 4.01235 7.54642 4.44713 6.69241C4.7481 6.05734 5.13833 5.46858 5.60599 4.94398C6.16001 4.34736 6.8007 3.83754 7.50652 3.43167C7.8302 3.24114 8.16941 3.07832 8.52052 2.94495C9.04221 2.75721 9.59283 2.66258 10.1473 2.66537C10.6436 2.66652 11.1376 2.73375 11.6161 2.86528C12.5756 3.10049 13.433 3.64024 14.0599 4.40366C14.7204 5.22066 14.987 6.2477 15.2028 7.63978C15.2578 8.00193 15.2796 8.18879 15.2984 8.35248C15.31 8.46547 15.323 8.57846 15.352 8.77836C15.5273 9.97199 15.6142 10.5167 15.8141 10.9324C16.1038 11.5408 16.6572 11.8957 17.0208 12.1304C17.9131 12.7098 18.8344 12.788 19.4442 12.8416C19.6847 12.8633 19.9585 12.8749 20.2569 12.8749C20.703 12.8749 21.0739 12.8503 21.2115 12.8402C21.4853 12.8199 21.7141 12.7938 21.9358 12.7677C22.272 12.7223 22.6106 12.6972 22.9498 12.6924C23.3746 12.6921 23.7976 12.7491 24.2071 12.8619C25.0966 13.067 25.8989 13.547 26.5002 14.2337C27.1101 14.958 27.2738 15.6721 27.4795 16.5804C27.6193 17.187 27.707 17.8044 27.7417 18.4259C27.7764 19.1313 27.8242 20.0946 27.4925 21.2303C27.1727 22.2747 26.6228 23.234 25.8831 24.0377C25.1073 24.908 24.1734 25.6233 23.1309 26.1453C22.5393 26.4354 21.9207 26.6669 21.2839 26.8363C20.4006 27.0835 19.4935 27.2365 18.578 27.2926C18.1912 27.3216 17.8175 27.3361 17.4655 27.3361Z"
              fill="#26283D"
            />
            <path
              d="M17.4655 26.2265C17.1323 26.2265 16.8194 26.2134 16.5109 26.1859C16.1688 26.1658 15.8293 26.1139 15.4969 26.0309C14.8468 25.8412 14.2394 25.5279 13.7079 25.1082C12.924 24.5046 12.2698 23.7494 11.7842 22.8875C11.6016 22.5645 11.3873 21.9908 10.8701 20.6031C10.7484 20.2786 10.6586 20.0323 10.5804 19.8295C10.4008 19.3385 10.3269 19.1342 10.1618 18.7764C9.94359 18.2639 9.6768 17.7735 9.36504 17.3119C8.98218 16.7792 8.53831 16.293 8.04249 15.8633C7.84694 15.6852 7.74264 15.6026 7.59778 15.4882C7.41961 15.3433 7.19942 15.1738 6.75471 14.7842C6.03042 14.1497 5.76968 13.8672 5.53791 13.5703C5.19569 13.1552 4.92274 12.6875 4.7296 12.1854C4.41134 11.3736 4.36059 10.4814 4.58474 9.63882C4.75558 8.79295 5.04066 7.97424 5.43216 7.20521C5.68904 6.6581 6.02321 6.15074 6.42443 5.69869C6.89941 5.18449 7.44931 4.74497 8.05553 4.39498C8.32534 4.23625 8.6078 4.1001 8.90005 3.98793C9.29948 3.84394 9.72125 3.77185 10.1458 3.77499C10.5448 3.77727 10.9417 3.83182 11.3264 3.93723C12.0597 4.10871 12.7166 4.51577 13.1965 5.09609C13.6311 5.6422 13.8846 6.37663 14.1062 7.80492C14.1584 8.14389 14.1772 8.31048 14.1946 8.47127C14.2076 8.5944 14.2221 8.71753 14.254 8.93481C14.4336 10.1531 14.5321 10.8252 14.8146 11.4104C15.2492 12.2984 15.9734 12.7735 16.4182 13.056C17.5394 13.7803 18.6692 13.8817 19.3443 13.9425C19.6166 13.9672 19.9237 13.9788 20.2554 13.9788C20.7393 13.9788 21.142 13.9527 21.2941 13.9411C21.5838 13.9194 21.8387 13.8904 22.0604 13.8643C22.3529 13.8246 22.6474 13.8019 22.9425 13.7962C23.2691 13.7958 23.5942 13.8401 23.9087 13.9281C24.2796 14.028 25.0502 14.238 25.647 14.9421C26.0816 15.4592 26.2062 16.001 26.393 16.8252C26.5186 17.3689 26.5976 17.9223 26.6292 18.4795C26.661 19.14 26.7016 19.9614 26.422 20.9174C26.1469 21.8088 25.6746 22.6267 25.0401 23.3105C24.3582 24.0719 23.5378 24.6968 22.6224 25.1516C22.0928 25.4117 21.5387 25.6184 20.9681 25.7687C20.1555 25.9947 19.3214 26.1345 18.4795 26.1859C18.1333 26.2192 17.7929 26.2265 17.4655 26.2265Z"
              fill="#A59678"
            />
            <path
              d="M9.21149 4.83535C10.0328 4.53405 10.8513 4.74409 11.0946 4.80783C11.6453 4.93454 12.1384 5.24019 12.4969 5.67697C12.8199 6.07533 13.0212 6.69098 13.2211 7.95558C13.3124 8.5437 13.295 8.56978 13.366 9.07823C13.5529 10.3472 13.6644 11.1062 14.0077 11.816C14.548 12.9314 15.4563 13.521 15.9329 13.8267C17.2366 14.6726 18.5403 14.7885 19.2646 14.8537C19.9624 14.9053 20.663 14.9053 21.3607 14.8537C22.4718 14.7711 22.9121 14.6045 23.6712 14.8117C24.0058 14.9015 24.549 15.0579 24.9546 15.536C25.2443 15.8851 25.337 16.2603 25.5109 17.0324C25.625 17.5246 25.6967 18.0257 25.7253 18.5302C25.7571 19.169 25.7904 19.86 25.5543 20.6697C25.3157 21.4363 24.906 22.1387 24.3563 22.7238C23.7531 23.3969 23.027 23.9485 22.2168 24.3491C21.736 24.5834 21.2327 24.7683 20.7146 24.901C19.9602 25.1106 19.1857 25.2397 18.4042 25.2863C17.7992 25.3373 17.1911 25.3373 16.5862 25.2863C16.2975 25.2698 16.0109 25.2267 15.7301 25.1574C15.1926 24.9997 14.6912 24.7384 14.254 24.3882C13.5654 23.8599 12.9908 23.1976 12.5649 22.4413C12.4201 22.1806 12.1811 21.5476 11.7088 20.2801C11.2916 19.1603 11.2323 18.943 10.9845 18.3969C10.7451 17.8345 10.451 17.297 10.1067 16.7919C9.68854 16.2019 9.20233 15.6632 8.65814 15.1869C8.19314 14.7639 8.16417 14.8044 7.35442 14.0961C6.64751 13.4761 6.43168 13.2342 6.25205 13.0053C5.96697 12.6629 5.73947 12.2764 5.57846 11.8609C5.45202 11.5217 5.37825 11.1651 5.35973 10.8035C5.35202 10.4734 5.39004 10.1439 5.47272 9.82425C5.62675 9.05475 5.88556 8.31001 6.24191 7.61082C6.46342 7.13752 6.75173 6.69847 7.09802 6.3071C7.32105 6.06667 7.56328 5.84479 7.82231 5.64366C8.03967 5.47474 8.26863 5.3213 8.50748 5.18446C8.73275 5.05002 8.96813 4.93329 9.21149 4.83535V4.83535Z"
              fill="url(#paint1_linear)"
            />
            <defs>
              <radialgradient
                cx="0"
                cy="0"
                gradientTransform="translate(15 15) scale(28.521)"
                gradientUnits="userSpaceOnUse"
                id="paint0_radial"
                r="1"
              >
                <stop
                  stop-color="#746B7F"
                />
                <stop
                  offset="0.12"
                  stop-color="#615B6F"
                />
                <stop
                  offset="0.26"
                  stop-color="#514C60"
                />
                <stop
                  offset="0.37"
                  stop-color="#4B475B"
                />
                <stop
                  offset="0.44"
                  stop-color="#4F4B5F"
                />
                <stop
                  offset="0.51"
                  stop-color="#5A566C"
                />
                <stop
                  offset="0.59"
                  stop-color="#6D6881"
                />
                <stop
                  offset="0.67"
                  stop-color="#87829E"
                />
                <stop
                  offset="0.74"
                  stop-color="#A9A2C3"
                />
                <stop
                  offset="0.78"
                  stop-color="#BCB5D9"
                />
              </radialgradient>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="paint1_linear"
                x1="5.54949"
                x2="23.9884"
                y1="7.79624"
                y2="23.7393"
              >
                <stop
                  offset="0.05"
                  stop-color="#448491"
                />
                <stop
                  offset="0.15"
                  stop-color="#4A7986"
                />
                <stop
                  offset="0.33"
                  stop-color="#506E7B"
                />
                <stop
                  offset="0.52"
                  stop-color="#526A77"
                />
                <stop
                  offset="0.6"
                  stop-color="#56707C"
                />
                <stop
                  offset="0.7"
                  stop-color="#607F8B"
                />
                <stop
                  offset="0.82"
                  stop-color="#7099A3"
                />
                <stop
                  offset="0.94"
                  stop-color="#87BDC4"
                />
                <stop
                  offset="1"
                  stop-color="#94D1D6"
                />
              </lineargradient>
            </defs>
          </svg>
        </a>
        <div
          class="border-r-[1px] border-black"
        />
        <a
          class="ml-2"
          href="https://chanzuckerberg.com"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="w-8 h-8"
            fill="none"
            height="30"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Go to Chan Zuckerberg Initiative main website.
            </title>
            <path
              d="M14.6702 13.6813C13.6812 14.3407 12.5274 14.8352 10.879 14.8352C8.901 14.8352 7.41748 13.5165 7.25265 11.5385C7.25265 10.5495 7.58232 9.56046 8.24166 8.73628C8.901 7.9121 9.89001 7.58243 10.879 7.58243C11.3735 7.58243 12.0329 7.74727 12.3625 7.9121C12.3625 7.9121 12.1977 9.06595 12.1977 9.56046H13.8461L14.1757 7.08793L13.6812 6.75826C12.857 6.26375 11.868 5.93408 10.879 5.93408C9.3955 5.93408 7.91198 6.59342 6.92297 7.58243C5.93396 8.73628 5.43945 10.055 5.43945 11.5385C5.60429 12.8572 6.09879 14.011 7.0878 15C8.07681 15.8242 9.3955 16.3187 10.7142 16.3187C10.7142 16.3187 10.7142 16.3187 10.879 16.3187C11.7032 16.3187 12.6922 16.1539 13.6812 15.6594L14.8351 13.3517C14.8351 13.6813 14.8351 13.6813 14.6702 13.6813Z"
              fill="black"
            />
            <path
              d="M20.1099 19.6154L19.945 21.9231C19.945 22.7473 19.4505 23.5714 18.7912 23.9011C18.2967 24.2308 17.6374 24.2308 17.1429 23.9011C16.6483 23.5714 16.3187 23.0769 16.1538 22.5824C15.989 21.5934 16.4835 20.7692 17.1429 20.2747L28.1868 13.1868C28.3516 13.8462 28.3516 14.3407 28.3516 15C28.3516 22.4176 22.4176 28.3516 15 28.3516C7.58241 28.3516 1.64835 22.4176 1.64835 15C1.64835 7.58242 7.58241 1.64835 15 1.64835C20.6044 1.64835 25.2198 4.94506 27.3626 9.72527L28.6813 8.9011C26.3736 3.62637 21.0989 0 15 0C6.75824 0 0 6.75824 0 15C0 23.2418 6.75824 30 15 30C23.2417 30 30 23.2418 30 15C30 14.011 29.8351 13.1868 29.6703 12.1978C29.5055 11.5385 29.3406 10.8791 29.3406 10.7143L26.2088 12.6923L22.0879 15.3297C21.9231 14.8352 21.5934 14.1758 20.9341 13.8462C20.1099 13.5165 18.7912 13.6813 17.8022 14.3407C17.8022 14.3407 20.4396 9.06593 20.9341 8.24176V8.07692H20.7692H15.8242L15.6593 9.72527H16.4835H18.4615L14.5055 17.1429C14.5055 17.3077 14.6703 17.4725 14.6703 17.3077L16.8132 16.1538C17.8022 15.6593 19.4505 14.5055 20.2747 15.1648C20.4395 15.3297 20.6044 15.4945 20.6044 15.6593C20.6044 15.8242 20.6044 15.8242 20.4396 15.8242L16.3187 18.6264C15 19.6154 14.5055 21.0989 14.5055 22.4176C14.6703 23.5714 15.3297 24.5604 16.3187 25.0549C16.8132 25.3846 17.4725 25.5494 18.1319 25.5494C18.6264 25.5494 19.1209 25.3846 19.6154 25.2198C20.7692 24.5604 21.5934 23.4066 21.7582 21.9231L22.0879 18.1319L20.1099 19.6154Z"
              fill="black"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
